prediction_market_agent/agents/microchain_agent/nft_treasury_game/deploy_nft_treasury_game.py

```diff
@@ -71,6 +92,43 @@ def load(self) -> None:
 
         super().load()
 
+    def before_iteration_callback(self) -> None:
+        if self.agent.history and GameRoundEnd.GAME_ROUND_END_OUTPUT in str(
+            self.agent.history[-1]
+        ):
+            # Just sleep for a very long time if the last thing the agent did was being done with this game.
+            # That way he won't be doing anything until the game is reset.
+            logger.info("Agent is done with the game, sleeping.")
+            time.sleep(31_536_000)
+
+    def after_iteration_callback(self) -> None:
+        system_prompt = self.agent.history[0]
+
+        if (
+            not self.game_finished_detected
+            and get_nft_game_status() == NFTGameStatus.finished
+        ):
+            # Switch to more capable (but a lot more expensive) model so that the reflections are worth it.
+            self.agent.llm.generator.model = SupportedModel.gpt_4o.value
```

Comment:
gpt-4o-mini's didn't write any good system prompts, they were general, without any learnings and ditched even stuff that were already in the system prompt (about the game, etc.)

Reply:
So when the time comes and agent is meant to reflect on its past actions and update its system prompt, this is a hacky switch to a more powerful model.